packages/augur-tools/src/flash/generate-templates.ts

```diff
@@ -0,0 +1,84 @@
+import { ethers } from 'ethers';
+import * as fs from 'fs';
+import {
+  Template,
+  TemplateInput,
+  ValidationTemplateInputType,
+  TemplateInputType,
+} from '../templates-template';
+import { TEMPLATES } from '../templates-source';
+
+const templateString = '//##TEMPLATES##';
+const templateValidationString = '//##TEMPLATE_VALIDATIONS##';
+const templateArtifactsFile = '../augur-artifacts/src/templates.ts';
+const templateTemplateFile = './src/templates-template.ts';
+
+export const generateTemplateValidations = async () => {
+  let validations = {};
+  let newTemplates = JSON.parse(JSON.stringify(TEMPLATES));
+  const topCategories = Object.keys(newTemplates);
+  topCategories.map(c => addTemplates(newTemplates[c], validations));
+  const newTemplateValue = `export const TEMPLATES = ${JSON.stringify(
+    newTemplates
+  )}`;
+  const newValidation = `export const TEMPLATE_VALIDATIONS = ${JSON.stringify(
+    validations
+  )}`;
+
+  if (!fs.existsSync(templateTemplateFile))
+    return console.error(templateTemplateFile, 'does not exist');
+
+  const contents = fs.readFileSync(templateTemplateFile, 'utf8');
+
+  const setNewTemplatesValues = contents.replace(
+    templateString,
+    newTemplateValue
+  );
+  const setNewTemplateValidations = setNewTemplatesValues.replace(
+    templateValidationString,
+    newValidation
+  );
+
+  fs.writeFileSync(templateArtifactsFile, setNewTemplateValidations, 'utf8');
+};
+
+const addTemplates = (category, validations) => {
```

Comment:
This ought to have types instead of implicit any.

Reply:
added types